src/main.cpp

```diff
@@ -919,19 +920,31 @@ bool ContextualCheckTransaction(
                 }
 
                 // SaplingNotePlaintext::decrypt() checks note commitment validity.
-                if (!SaplingNotePlaintext::decrypt(
+                auto encPlaintext = SaplingNotePlaintext::decrypt(
```

Comment:
Note that this will reject the plaintext based on `plaintext_version_is_valid` internally, so the logic that later checks `if (canopyActive != (encPlaintext->get_leadbyte() == 0x02)) {` is essentially overriding the grace period logic of `plaintext_version_is_valid`. Perhaps this is worth mentioning in a comment.

Reply:
thanks for pointing this out. we had discussed this earlier: https://github.com/zcash/zcash/pull/4578#discussion_r445061646

i'm adding in this comment below: 
```C++
// This check compels miners to switch to the new plaintext version
// and overrides the grace period in plaintext_version_is_valid()
if (canopyActive != (encPlaintext->get_leadbyte() == 0x02)) {
```